src/Application/Actions/Donations/Update.php

```diff
@@ -194,35 +195,32 @@ private function addData(Donation $donation, HttpModels\Donation $donationData):
 
         if ($donation->getPsp() === 'stripe') {
             try {
-                $this->stripeClient->paymentIntents->update($donation->getTransactionId(), [
-                    'amount' => $donation->getAmountFractionalIncTip(),
-                    'currency' => strtolower($donation->getCurrencyCode()),
-                    'metadata' => [
-                        /**
-                         * Note that we don't re-set keys that can't change, like `charityId`.
-                         * See the counterpart in {@see Create::action()} too.
-                         */
-                        'coreDonationGiftAid' => $donation->hasGiftAid(),
-                        'feeCoverAmount' => $donation->getFeeCoverAmount(),
-                        'matchedAmount' => $donation->getFundingWithdrawalTotal(),
-                        'optInCharityEmail' => $donation->getCharityComms(),
-                        'optInTbgEmail' => $donation->getTbgComms(),
-                        'salesforceId' => $donation->getSalesforceId(),
-                        'stripeFeeRechargeGross' => $donation->getCharityFeeGross(),
-                        'stripeFeeRechargeNet' => $donation->getCharityFee(),
-                        'stripeFeeRechargeVat' => $donation->getCharityFeeVat(),
-                        'tbgTipGiftAid' => $donation->hasTipGiftAid(),
-                        'tipAmount' => $donation->getTipAmount(),
-                    ],
-                    // See https://stripe.com/docs/connect/destination-charges#application-fee
-                    // Update the fee amount incase the final charge was from
-                    // a Non EU / Amex card where fees are varied.
-                    'application_fee_amount' => $donation->getAmountToDeductFractional(),
-                    // Note that `on_behalf_of` is set up on create and is *not allowed* on update.
-                ]);
-
-                if ($donationData->autoConfirmFromCashBalance) {
-                    $this->stripeClient->paymentIntents->confirm($donation->getTransactionId());
+                $this->updatePaymentIntent($donation);
+            } catch (RateLimitException $exception) {
+                if ($exception->getStripeCode() !== 'lock_timeout') {
+                    throw $exception; // Only re-try when object level lock failed.
+                }
+
+                $this->logger->info(sprintf(
+                    'Stripe lock "rate limit" hit while updating payment intent for donation %s – retrying in 1s...',
+                    $donation->getId(),
+                ));
+                sleep(1);
```

Comment:
this will work for now, but we may want to change it soon, e.g. to make the sleep duration injected so we don't have to wait a second when running the test for this, or to use a queue system to do the updates.